### PR TITLE
Add util functions, callback response object

### PR DESCRIPTION
**Pull request by dsquier**
- Adds response object to report callback
- Adds two util functions: `audit()`, `hash()` with associated tests
-- util.audit adds an object to an array that is returned upon
completion
-- util.hash returns a SHA256 hash using node-object-hash
- Replaces use of var with const/let where appropriate
- Removes lodash require() from db.js (not used in file)
- Upgrades node, async, lodash, nano dependency versions
- Bumped version from 0.2.2 -> 0.3.0

### CODE REVIEW

File: test/buster.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+config["Couchtato Tests"] = {
+    rootPath: "../",
+    environment: "node", // or "node"
+    sources: [
+        "lib/*.js"
+    ],
+    tests: [
+        "test/*.js"
+    ]
+}
```
Comment (cliffano) [inline test/buster.js]: Missing semicolon here, would cause lint failure.

File: README.md [hunk 2]
```diff
@@ -1,12 +1,10 @@
 <img align="right" src="https://raw.github.com/cliffano/couchtato/master/avatar.jpg" alt="Avatar"/>
 
-[![Build Status](https://img.shields.io/travis/cliffano/couchtato.svg)](http://travis-ci.org/cliffano/couchtato)
+<!--[![Build Status](https://img.shields.io/travis/cliffano/couchtato.svg)](http://travis-ci.org/cliffano/couchtato)
```
Comment (cliffano) [inline README.md]: Is this commented out by mistake?

**Comment by dsquier** [comment]
Sure thing-- I'll update the PR with these fixes. Thanks for the code review!

File: README.md (hunk 2 above)
Comment (dsquier) [inline README.md]: Yeah, this shouldn't have been commented out. Will adjust also.

**Comment by cliffano** [comment]
@dsquier Thanks for the PR, much appreciated.

**Pull request merged**